plugin-core/src/main/java/appland/telemetry/TelemetryProperties.java

```diff
@@ -0,0 +1,62 @@
+package appland.telemetry;
+
+import appland.AppMapPlugin;
+import appland.utils.GsonUtils;
+import com.google.gson.annotations.SerializedName;
+import com.intellij.openapi.application.ApplicationInfo;
+import com.intellij.openapi.util.SystemInfo;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class TelemetryProperties {
+    public static Map<String, String> getProperties(boolean includeUsername) {
+        var properties = new HashMap<String, String>();
+        properties.put("extname", AppMapPlugin.getDescriptor().getPluginId().getIdString());
+        properties.put("extversion", AppMapPlugin.getDescriptor().getVersion());
+        properties.put("ide", ApplicationInfo.getInstance().getFullApplicationName());
+        properties.put("ideversion", ApplicationInfo.getInstance().getFullVersion());
+        properties.put("os", SystemInfo.OS_NAME);
+        properties.put("osversion", SystemInfo.OS_VERSION);
+        properties.put("jvmversion", System.getProperty("java.version"));
+        properties.put("product", ApplicationInfo.getInstance().getBuild().getProductCode());
+        properties.put("source", "JetBrains");
+
+        if (includeUsername) {
+            try {
+                properties.put("username", System.getProperty("user.name"));
+            } catch (Exception e) {
+                // ignore
+            }
+        }
+        return properties;
+    }
+
+    public static @NotNull String getPropertiesAsJson() {
+        return GsonUtils.GSON.toJson(new CliProperties(getProperties(false)));
+    }
+
+    /**
+     * Class to serialize for the CLI env variable.
+     * It's re-serializing from a map to apply the @SerializedName annotations.
+     */
+    private static class CliProperties {
+        @SerializedName("extname")
+        final String extName;
+        @SerializedName("extversion")
+        final String extVersion;
+        @SerializedName("ide")
+        final String ide;
+        @SerializedName("ideversion")
+        final String ideVersion;
+
+        public CliProperties(@NotNull Map<String, String> properties) {
+            // Note these are the only properties we need for the CLI
+            this.extName = properties.get("extname");
```

Comment:
The property names are hard-coded twice in `TelemetryProperties` (here and above).
Consider extracting the name as a constant or to avoid the duplicated names in another way.